### ISSUE TITLE
[BUG REPORT] 

### ISSUE THREAD

**Issue by emanueledonati** [gitmylo/audio-webui#61]
I'm trying to run the software but I get the following, what can I do?
thanks in advance for your support!
E.

C:\audio-webui\audio-webui>run
Checking installs and venv + autodebug checks
Python version:  3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
activating venv
Checking installs and venv + autodebug checks
Python version:  3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
Installed PyTorch!

    ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
    █░▄▄▀██░██░█▄▄░▄▄██░▄▄▄░██░▄▄▀██░▄▄▄██░▄▄▀██░██░██░▄▄░██
    █░▀▀░██░██░███░████░███░██░██░██░▄▄▄██░▄▄▀██░██░██░█▀▀██
    █░██░██▄▀▀▄███░████░▀▀▀░██░▀▀░██░▀▀▀██░▀▀░██▄▀▀▄██░▀▀▄██
    ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
Install failed!
STDOUT:
Looking in indexes: https://download.pytorch.org/whl/cu117
Collecting torch

STDERR:
ERROR: Exception:
Traceback (most recent call last):
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_internal\cli\base_command.py", line 160, in exc_logging_wrapper
    status = run_func(*args)
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_internal\cli\req_command.py", line 247, in wrapper
    return func(self, options, args)
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_internal\commands\install.py", line 400, in run
    requirement_set = resolver.resolve(
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_internal\resolution\resolvelib\resolver.py", line 92, in resolve
    result = self._result = resolver.resolve(
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_vendor\resolvelib\resolvers.py", line 481, in resolve
    state = resolution.resolve(requirements, max_rounds=max_rounds)
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_vendor\resolvelib\resolvers.py", line 348, in resolve
    self._add_to_criteria(self.state.criteria, r, parent=None)
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_vendor\resolvelib\resolvers.py", line 172, in _add_to_criteria
    if not criterion.candidates:
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_vendor\resolvelib\structs.py", line 151, in __bool__
    return bool(self._sequence)
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_internal\resolution\resolvelib\found_candidates.py", line 155, in __bool__
    return any(self)
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_internal\resolution\resolvelib\found_candidates.py", line 143, in <genexpr>
    return (c for c in iterator if id(c) not in self._incompatible_ids)
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_internal\resolution\resolvelib\found_candidates.py", line 47, in _iter_built
    candidate = func()
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_internal\resolution\resolvelib\factory.py", line 206, in _make_candidate_from_link
    self._link_candidate_cache[link] = LinkCandidate(
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_internal\resolution\resolvelib\candidates.py", line 297, in __init__
    super().__init__(
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_internal\resolution\resolvelib\candidates.py", line 162, in __init__
    self.dist = self._prepare()
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_internal\resolution\resolvelib\candidates.py", line 231, in _prepare
    dist = self._prepare_distribution()
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_internal\resolution\resolvelib\candidates.py", line 308, in _prepare_distribution
    return preparer.prepare_linked_requirement(self._ireq, parallel_builds=True)
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_internal\operations\prepare.py", line 491, in prepare_linked_requirement
    return self._prepare_linked_requirement(req, parallel_builds)
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_internal\operations\prepare.py", line 536, in _prepare_linked_requirement
    local_file = unpack_url(
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_internal\operations\prepare.py", line 166, in unpack_url
    file = get_http_url(
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_internal\operations\prepare.py", line 107, in get_http_url
    from_path, content_type = download(link, temp_dir.path)
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_internal\network\download.py", line 134, in __call__
    resp = _http_get_download(self._session, link)
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_internal\network\download.py", line 117, in _http_get_download
    resp = session.get(target_url, headers=HEADERS, stream=True)
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_vendor\requests\sessions.py", line 600, in get
    return self.request("GET", url, **kwargs)
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_internal\network\session.py", line 518, in request
    return super().request(method, url, *args, **kwargs)
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_vendor\requests\sessions.py", line 587, in request
    resp = self.send(prep, **send_kwargs)
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_vendor\requests\sessions.py", line 701, in send
    r = adapter.send(request, **kwargs)
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_vendor\cachecontrol\adapter.py", line 48, in send
    cached_response = self.controller.cached_request(request)
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_vendor\cachecontrol\controller.py", line 155, in cached_request
    resp = self.serializer.loads(request, cache_data, body_file)
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_vendor\cachecontrol\serialize.py", line 95, in loads
    return getattr(self, "_loads_v{}".format(ver))(request, data, body_file)
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_vendor\cachecontrol\serialize.py", line 186, in _loads_v4
    cached = msgpack.loads(data, raw=False)
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_vendor\msgpack\fallback.py", line 123, in unpackb
    unpacker.feed(packed)
  File "C:\audio-webui\audio-webui\venv\lib\site-packages\pip\_vendor\msgpack\fallback.py", line 381, in feed
    self._buffer.extend(view)
MemoryError

[notice] A new release of pip available: 22.3.1 -> 23.1.2
[notice] To update, run: python.exe -m pip install --upgrade pip

Please read the error above carefully.
If you are unsure, please create an issue at https://github.com/gitmylo/audio-webui/issues.
When creating an issue, please include your full autodebug message.
Exit code: 2

**Comment by gitmylo** [comment]
Does this happen every time you run?
What are your system specs?

**Comment by emanueledonati** [comment]
11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz   1.80 GHz
16,0 GB (15,7 GB free)
Sistema operativo a 64 bit, processore basato su x64
Windows 11 Home
Versione	22H2
372Gb free on HD

**Comment by gitmylo** [comment]
Did you try running it again? Does it happen every time?

Maybe you're running out of RAM during loading from pip cache.

Maybe increasing your pagefile helps? Pagefile is used when you run out of ram. Which seems to be the issue here.

Here is an [online guide](https://www.thewindowsclub.com/increase-page-file-size-virtual-memory-windows) explaining how to increase your page file.

You could also run with `--no-cache-dir` in pip, but i would need to add a new argument for that. Unless you want to set that yourself (in `setup_tools/magicinstaller/requirement.py#68`).


## In short, you have 2 options:
1. [Increase page file size](https://www.thewindowsclub.com/increase-page-file-size-virtual-memory-windows)
2. Add `--no-cache-dir` to pip command manually (in `setup_tools/magicinstaller/requirement.py#68`)

**Comment by gitmylo** [comment]
I will close this issue, if you're still having issues, make sure you did the steps above. If they don't work, just send another message here, i will see them even if the issue is closed. I just don't think it's useful to keep it open for now.
